### PR TITLE
1767 different visual appearances

### LINKED ISSUE #1767: Different visual appearances for enums
Currently the user settings editor displays enums in comboboxes.

It should be possible to display a group of options as [segmented button](https://sap.github.io/fundamental-styles/?path=/docs/components-button--primary) too.

### PR DESCRIPTION
**Pull request by legteodav**
**Related issue(s)**
`Resolves #1767 `

### CODE REVIEW

File: core/src/UserSettingsEditor.html [hunk 1]
```diff
@@ -59,6 +59,18 @@
         userSettings[userSettingsGroupSchema[0]][key] = option;
     }
 
+    function updateEnumButton(key, option) {
+      document.querySelectorAll('.enum-buttons-container-'+ key  +' button').forEach(button => {
+        let buttonTestid = button.getAttribute('data-testid');
+        let optionTestid = 'lui-us-enum_button_'+key+'_'+option;
```
Comment (stanleychh) [inline core/src/UserSettingsEditor.html:65]:
```suggestion
        const optionTestid = `lui-us-enum_button_${key}_option`;
```

File: core/src/UserSettingsEditor.html [hunk 2]
```diff
@@ -59,6 +59,18 @@
         userSettings[userSettingsGroupSchema[0]][key] = option;
     }
 
+    function updateEnumButton(key, option) {
+      document.querySelectorAll('.enum-buttons-container-'+ key  +' button').forEach(button => {
+        let buttonTestid = button.getAttribute('data-testid');
```
Comment (stanleychh) [inline core/src/UserSettingsEditor.html:64]: I am not a fan of using test id here. Test Id is for cypress only. Maybe use custom data attributes instead?

**Review by stanleychh — CHANGES_REQUESTED** [review]

File: core/src/UserSettingsEditor.html (hunk 2 above)
Comment (stanleychh) [reply core/src/UserSettingsEditor.html:64]: oh my bad, I was not to see it clearly. It already used data attribute here.

Comment (stanleychh) [reply core/src/UserSettingsEditor.html:64]: Or give it an `id` and use it.

Comment (stanleychh) [reply core/src/UserSettingsEditor.html:64]: @legteodav , I would suggest to have `id` attribute instead. Keep test id for cypress only. How do you think?

Comment (legteodav) [reply core/src/UserSettingsEditor.html]: Just changed it..... (sorry I forgot...)

File: core/src/UserSettingsEditor.html [hunk 3]
```diff
@@ -1,183 +1,265 @@
...
 
-    function setExpandedState(self, value) {
-        const indicatorBtn = self.querySelector('.fd-input-group__button');
-        const optionsPopOver = self.parentNode.querySelector('.fd-popover__body');
-        self.setAttribute('aria-expanded', value);
-        optionsPopOver.setAttribute('aria-hidden', !value);
-        indicatorBtn.setAttribute('aria-expanded', value);
+  function toggleOptions(event, editable) {
+    if (editable || editable === undefined) {
+      console.log('test');
```
Comment (stanleychh) [inline core/src/UserSettingsEditor.html:54]:
```suggestion
```

File: core/src/UserSettingsEditor.html [hunk 4]
```diff
@@ -1,183 +1,265 @@
...
+  function updateComboBox(key, option) {
+    userSettings[userSettingsGroupSchema[0]][key] = option;
+  }
 
+  function updateEnumButton(key, option) {
+    document
+      .querySelectorAll('.enum-buttons-container-' + key + ' button')
+      .forEach(button => {
+        let buttonTestid = button.getAttribute('id');
+        const optionTestid = `lui-us-enum_button_${key}_option`;
```
Comment (stanleychh) [inline core/src/UserSettingsEditor.html:79]:
```suggestion
        const optionId = `lui-us-enum_button_${key}_option`;
```

File: core/src/UserSettingsEditor.html [hunk 5]
```diff
@@ -1,183 +1,265 @@
...
+        buttonTestid === optionTestid
```
Comment (stanleychh) [inline core/src/UserSettingsEditor.html:80]:
```suggestion
        buttonId === optionId
```

**Review by JohannesDoberer — APPROVED** [review]
LGTM! Good Job 👍

**Review by stanleychh — APPROVED** [review]

**Pull request merged**